WordPressKit/HTTPRequestBuilder.swift

```diff
@@ -156,13 +158,15 @@ final class HTTPRequestBuilder {
             request.addValue(value, forHTTPHeaderField: header)
         }
 
-        if encodeMultipartForm {
-            let encoded = try self.encodeMultipartForm(request: &request, forceWriteToFile: false)
-            switch encoded {
-            case let .left(data):
-                request.httpBody = data
-            case let .right(url):
-                request.httpBodyStream = InputStream(url: url)
+        if encodeBody {
+            let body = try encodeMultipartForm(request: &request, forceWriteToFile: false) ?? encodeXMLRPC(request: &request, forceWriteToFile: false)
+            if let body {
+                switch body {
+                case let .left(data):
+                    request.httpBody = data
+                case let .right(url):
+                    request.httpBodyStream = InputStream(url: url)
+                }
```

Comment:
Maybe a bit too compact, but what do you think of 

```suggestion
        if encodeBody, let body = try encodeMultipartForm(request: &request, forceWriteToFile: false) ?? encodeXMLRPC(request: &request, forceWriteToFile: false) {
            switch body {
            case let .left(data):
                request.httpBody = data
            case let .right(url):
                request.httpBodyStream = InputStream(url: url)
            }
```

Reply:
I tried that, but it looks like many things are squeezed together, so I went with current version that has a few more lines.